servlet.py

```diff
@@ -797,6 +797,91 @@ def get(self):
         else:
             self.send_error(400, explanation='That mode is not installed')
 
+class DictionaryLookupHandler(BaseHandler):
+    def getPipeCmds(self, l1, l2):
+        if (l1, l2) not in self.pipeline_cmds:
+            mode_path = self.pairs['%s-%s' % (l1, l2)]
+            self.pipeline_cmds[(l1, l2)] = translation.parseModeFile(mode_path)
+        return self.pipeline_cmds[(l1, l2)]
+
+    def getPairOrError(self, langpair):
```

Comment:
Um, not all classes that extend `BaseHandler` use it. Do you suggest putting it at the `BaseHandler` level so that all the classes which extend it have this method?